dipy/core/geometry.py

```diff
@@ -126,7 +126,7 @@ def cart2sphere(x, y, z):
        azimuth angle
     '''
     r = np.sqrt(x * x + y * y + z * z)
-    theta = np.arccos(z / r)
+    theta = np.arccos(z / r, where=r > 0)
```

Comment:
it seems we still have the warning. That's weird... So,  I just tested another option: `np.arccos(np.divide(z, r, 
 where=r > 0))`
This solution seems to totally remove the warning.

Reply:
Let's see...